frontend/src/components/form/base/__tests__/EDatePicker.spec.js

```diff
@@ -0,0 +1,122 @@
+import Vuetify from 'vuetify'
+
+import i18n from '@/plugins/i18n'
+import formBaseComponents from '@/plugins/formBaseComponents'
+import { vueMoment, moment } from '@/plugins/vueMoment'
+
+import { mount as mountComponent } from '@vue/test-utils'
+import EDatePicker from '../EDatePicker'
+
+Vue.use(Vuetify)
+Vue.use(formBaseComponents)
+Vue.use(vueMoment, { moment })
+
+describe('An EDatePicker', () => {
+  let vuetify
+
+  const DATE_1 = '2020-03-01'
+  const DATE_2 = '2015-12-31'
+  const INVALID_DATE_1 = 'some date'
+  const INVALID_DATE_2 = '2026-13-01'
+
+  const localeData = [
+    [
+      'de',
+      {
+        date_1: '01.03.2020',
+        date_2: '31.12.2015',
+        date_3: '24.03.2020'
+      }
+
+    ],
+    [
+      'en',
+      {
+        date_1: '03/01/2020',
+        date_2: '12/31/2015',
+        date_3: '03/24/2020'
+      }
+
+    ]
+  ]
+
+  const flushPromises = () => new Promise((resolve) => setImmediate(resolve))
+
+  const waitForDebounce = () => new Promise((resolve) => setTimeout(resolve, 110))
+
+  const mount = (options) => mountComponent(EDatePicker, { vuetify, i18n, ...options })
+
+  describe.each(localeData)('in locale %s', (locale, data) => {
+    beforeEach(() => {
+      i18n.locale = locale
+      vuetify = new Vuetify()
+    })
+
+    test('renders', async () => {
+      const wrapper = mount({
+        propsData: {
+          value: DATE_1
+        }
+      })
+      await flushPromises()
+      expect(wrapper.find('input[type=text]').element.value).toBe(data.date_1)
+    })
+
+    test('updates v-model when the value changes', async () => {
+      const wrapper = mountComponent({
+        data: () => ({ date: DATE_1 }),
+        template: '<div><e-date-picker v-model="date"></e-date-picker></div>',
+        components: { 'e-date-picker': EDatePicker }
+      }, {
+        vuetify,
+        i18n
+      })
+      expect(wrapper.vm.date).toBe(DATE_1)
+      const inputSpy = jest.fn()
+      wrapper.findComponent(EDatePicker).vm.$on('input', (event) => inputSpy(event))
+      const input = wrapper.find('input[type=text]')
+      await input.setValue(data.date_2)
+      await waitForDebounce()
+      expect(inputSpy).toBeCalledTimes(1)
+      expect(inputSpy).toBeCalledWith(DATE_2)
+      expect(wrapper.vm.date).toBe(DATE_2)
+    })
+
+    test('validates the input', async () => {
+      const wrapper = mount({
+        propsData: {
+          value: DATE_1
+        }
+      })
+      const input = wrapper.find('input[type=text]')
+      await input.setValue(INVALID_DATE_1)
+      await waitForDebounce()
+      expect(wrapper.text()).toContain('invalid format')
+      await input.setValue(INVALID_DATE_2)
+      await waitForDebounce()
+      expect(wrapper.text()).toContain('invalid month')
+    })
+
+    test('updates its value when a date is picked', async () => {
+      // prevent "[Vuetify] Unable to locate target [data-app]" warnings
+      document.body.setAttribute('data-app', 'true')
```

Comment:
Why is this only needed in this test?

Reply:
I guess because it's the only place where the actual picker modal is rendered, and it looks for a container to attach to with this property.